libbeat/management/management.go

```diff
@@ -82,6 +83,14 @@ type Manager interface {
 
 	// CheckRawConfig check settings are correct before launching the beat.
 	CheckRawConfig(cfg *common.Config) error
+
+	// RegisterAction registers action handler with the client
+	RegisterAction(action client.Action)
+	// UnregisterAction unregisters action handler with the client
+	UnregisterAction(action client.Action)
+
+	// SetPayload sets the client payload
+	SetPayload(map[string]interface{})
```

Comment:
As this interface describes how input/Beats developers interact with the Agent: What exactly is the payload, what can/users do with it? Am I required to set it, or is it better to ignore `SetPayload`?

My understanding is that Endpoint uses the payload in order to pass some 'private monitoring' data via Agent to Kibana. I don't see why we need this on the Beats side as well, though.

Reply:
The payload was something that was not exposed for the beat to set. 
In the case of osquerybeat work we need to leverage the payload to communicate to the agent the version of osqueryd. 
Also using payload to communicate to the agent what inputs the application/beat is subscribed to.
This allows the agent to not care and not parse the the config details of the particular app/beat. 
Upon connection to the agent the beat subscribes to the actions on a receiving gRCP side and communicates back to the agent what "input types" are supported by the app. 
More details are in the original PR: https://github.com/elastic/beats/pull/24456

Reply:
Is this specific to osquerybeat?

Why not have a proper interface/type for the payload if it is bound to actual functionality in Agent?
If other Beats use a Payload, will it be used for similar or completely different functionality?

Reply:
It is generic just was not exposed, so the beats could not set their custom payload. 
If you look at the current status updates, they pass nil with the status update payload 
```
cm.client.Status(proto.StateObserved_STOPPING, "Stopping", nil)
```

@blakerouse mentioned before that there is already existing payload we could leverage to pass the custom payload over gRPC back to the agent. So I leveraged what we already have in place. Open for other suggestions.

Reply:
I guess I'm missing context here. 

1. Why not have a proper interface/type for the payload if it is bound to actual functionality in Agent?
2. If other Beats use a Payload, will it be used for similar or completely different functionality?

Reply:
> I guess I'm missing context here.
> 
> 1. Why not have a proper interface/type for the payload if it is bound to actual functionality in Agent?
We already have the payload that can be used. So I might be missing the bigger picture:
It's possible that I'm leveraging something the way it should be used.
Not sure why we need another payload. What was the original purpose of this existing payload? and why it can't be 
used for this particular purpose.

> 2. If other Beats use a Payload, will it be used for similar or completely different functionality?
It's whatever payload they want to send back that gets passed with the checkin events back to the server.

So there are two things that I use the payload for:
1. passing the osqueryd binary version with check in to the server
2. passing the input_types array back to the elastic agent so it know where to dispatch the actions

Is this the right approach? @blakerouse

Reply:
@aleksmaus I think in this context you are miss using the payload. The payload has always been opaque to Elastic Agent. Using it for registering the input_types to the Elastic Agent feels more like a hack than the proper use of payload.

I don't think it is correct behavior that the input types are told to the Agent from the beat anyway. I think it would be better for this to be determined by the Agent already. Being that the specification file in Agent already has to code in the input type that is allowed to a beat, then that should be used instead.

Basically removing the need for payload to be used for the input_types. I do think you are using it correctly for the version of osquery.